src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockPhysicsListener.java

```diff
@@ -112,4 +115,34 @@ public void onBucketUse(PlayerBucketEmptyEvent e) {
             e.setCancelled(true);
         }
     }
+
+    @EventHandler
+    public void onBlockUpdate(BlockPhysicsEvent event) {
+        if (!Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_18)) {
+            // Method BlockData#isSupported is 1.18+
+            return;
+        }
+
+        Block block = event.getBlock();
+
+        // Listen for collapsing sensitive slimefun blocks
+        if (BlockStorage.hasBlockInfo(block) && !Slimefun.getTickerTask().isDeletedSoon(block.getLocation()) && !block.getBlockData().isSupported(block)) {
+            SlimefunItem sfItem = BlockStorage.check(block);
+
+            if (sfItem != null && !sfItem.useVanillaBlockBreaking()) {
+                // Drop items from surrounding sensitive blocks but don't drop from supporting block
+                event.setCancelled(true);
+                block.setType(Material.AIR, true);
+
+                for (ItemStack drop : sfItem.getDrops()) {
+                    if (drop != null && !drop.getType().isAir()) {
+                        block.getWorld().dropItemNaturally(block.getLocation(), drop);
+                    }
+                }
+
+                // Fixes #2944 - Don't forget to clear the Block Data
+                BlockStorage.clearBlockInfo(block);
```

Comment:
We are also missing the call to block break handler, which could lead to massive amounts of unintended behavior

Reply:
Ah, good catch. Done